x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -381,12 +382,22 @@ export class ESSearchSource extends AbstractESSource {
     return !!sortField && !!sortOrder;
   }
 
-  async getGeoJsonWithMeta(layerName, searchFilters, registerCancelCallback) {
+  async getGeoJsonWithMeta(layerName, searchFilters, prevMeta, registerCancelCallback) {
+    const indexPattern = await this.getIndexPattern();
+
+    const indexSettings = prevMeta.indexSettings
+      ? prevMeta.indexSettings
+      : await loadIndexSettings(indexPattern.title);
```

Comment:
not sure if this is important. but would it make sense to cache this per user-session? this call precedes every data-fetch, and in 99% of use-cases the results will remain identical.